source/common/ssl/utility.cc

```diff
@@ -101,5 +105,43 @@ SystemTime Utility::getExpirationTime(const X509& cert) {
   return std::chrono::system_clock::from_time_t(days * 24 * 60 * 60 + seconds);
 }
 
+std::vector<uint8_t> Utility::getSha256Digest(const Buffer::Instance& buffer) {
+  std::vector<uint8_t> digest(SHA256_DIGEST_LENGTH);
+  EVP_MD_CTX ctx;
+  auto rc = EVP_DigestInit(&ctx, EVP_sha256());
+  RELEASE_ASSERT(rc == 1, "Failed to init digest context");
+  const auto num_slices = buffer.getRawSlices(nullptr, 0);
+  STACK_ARRAY(slices, Buffer::RawSlice, num_slices);
+  buffer.getRawSlices(slices.begin(), num_slices);
+  for (const auto& slice : slices) {
+    rc = EVP_DigestUpdate(&ctx, slice.mem_, slice.len_);
+    RELEASE_ASSERT(rc == 1, "Failed to update digest");
+  }
+  unsigned int digest_length;
+  rc = EVP_DigestFinal(&ctx, digest.data(), &digest_length);
+  RELEASE_ASSERT(rc == 1, "Failed to finalize digest");
+  RELEASE_ASSERT(digest_length == SHA256_DIGEST_LENGTH, "Digest length mismatch");
```

Comment:
I know we check this in a few other places, but it turns out this isn't necessary.